File: public/examples/integrations/mcp-servers/clickup_api/get_list_tasks_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "ClickupApi.GetListTasks"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    'list_identifier': 12345,
    'custom_task_type_filters': [0, 1],
    'display_tasks_in_reverse_order': True,
    'filter_by_assignees': [101, 102],
    'include_archived_tasks': False,
    'order_by_field': 'due_date',
    'page_number_to_fetch': 0
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(json.dumps(response.output.value, indent=2))

File: public/examples/integrations/mcp-servers/clickup_api/get_list_tasks_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "ClickupApi.GetListTasks";

// Start the authorization process
const authResponse = await client.tools.authorize({
  tool_name: TOOL_NAME,
  user_id: USER_ID,
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "list_identifier": 12345,
  "custom_task_type_filters": [
    0,
    1
  ],
  "display_tasks_in_reverse_order": true,
  "filter_by_assignees": [
    101,
    102
  ],
  "include_archived_tasks": false,
  "order_by_field": "due_date",
  "page_number_to_fetch": 0
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
